sound/soc/sof/sof-priv.h

```diff
@@ -283,6 +283,7 @@ struct snd_sof_dsp_ops {
 	void (*machine_unregister)(struct snd_sof_dev *sdev,
 				   void *pdata); /* optional */
 	struct snd_soc_acpi_mach * (*machine_select)(struct snd_sof_dev *sdev); /* optional */
+	struct snd_sof_of_mach * (*machine_info_select)(struct snd_sof_dev *sdev); /* optional */
```

Comment:
can we name this as `of_machine_select`?

In a followup we could rename the existing `machine_select` to `acpi_machine_select` and then modify the wrapper in sound/soc/sof/ops.h to:
```c
static inline void *snd_sof_machine_select(struct snd_sof_dev *sdev)
{
	if (!sof_ops(sdev))
		return NULL;

	if (sof_ops(sdev)->acpi_machine_select)
		return sof_ops(sdev)->acpi_machine_select(sdev);
	else if (sof_ops(sdev)->of_machine_select)
		return sof_ops(sdev)->of_machine_select(sdev);

	return NULL;
}
```
It might make things a bit easier to read?